### PR TITLE
Add scripting API to access briefing color tags

### PR DESCRIPTION
**Pull request by asarium**

### CODE REVIEW

**Comment by asarium** [comment]
I found another issue with the new UI code so I'll just include that fix here as well.

File: code/scripting/api/libs/ui.cpp [hunk 1]
```diff
@@ -386,6 +390,46 @@ ADE_LIB_DERIV(l_UserInterface_CmdBrief,
 	"API for the new UI system. This should not be used by other code and may be removed in the future!",
 	l_UserInterface);
 
+ADE_VIRTVAR(ColorTags,
+	l_UserInterface_CmdBrief,
+	nullptr,
+	"The available tagged colors",
+	ade_type_map("string", "color"),
+	"A apping from tag string to color value")
```
Comment (Goober5000) [inline code/scripting/api/libs/ui.cpp:398]: I assume this should be "A mapping"

File: code/scripting/api/objs/color.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+			CLAMP(newVal, 0.0f, 255.0f);
+		}
+
+		clr->*member = static_cast<ubyte>(newVal);
+	}
+
+	return ade_set_args(L, "i", clr->*member);
+}
+
+ADE_VIRTVAR(Red,
```
Comment (Goober5000) [inline code/scripting/api/objs/color.cpp:31]: I suggest changing this to `RedValue` or something similar; and the same with Green and Blue.  A scripter might think that `color.Red` is a factory method that returns a Color object set to pure red.

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Looks good overall but I had a few comments.

File: code/scripting/api/libs/ui.cpp (hunk 1 above)
Comment (asarium) [reply code/scripting/api/libs/ui.cpp]: Correct!

File: code/scripting/api/objs/color.cpp (hunk 2 above)
Comment (asarium) [reply code/scripting/api/objs/color.cpp:31]: In Lua, the `color` "class" is not available to scripts and `color.Red` is not a valid expression so this should not be an issue.

Comment (Goober5000) [reply code/scripting/api/objs/color.cpp:31]: Well it's not my preference, but I'll approve it.

**Review by Goober5000 — APPROVED** [review]

**Pull request merged**